Rewrite screenshot_render + add screenshot_cubemap command + misc changes

Berthalamew opened the pull request:
- screenshot render function rewritten to make sure the screenshot render target is only created if we're taking a screenshot, rather than every frame
- screenshots are currently broken in split screen (legacy bug), we can fix it at some point in the future?
- screenshot_cubemap takes in the name of the tif file as a parameter which will be saved in the main game directory

Berthalamew committed:
rename time_globals::get_game_time to get_game_time_ticks

Berthalamew committed:
move commands_map out of header into cpp for ServerConsole

- prevents duplication throughout translation units
- adjust includes in log header
- make is_valid memeber function const in h2log class

Berthalamew committed:
move rasterizer present hook

- adjust bitmap format enum and bitmap_data struct names

Berthalamew committed:
remove unused bWait condition

- removed since it's an sleeps forever and the bWait parameter is never set to true when called from h2v

Berthalamew committed:
remove patchcall for object_get_model_node_data that's no longer needed

Berthalamew committed:
misc changes

- remove unneeded check in scenario_tags_load
- adjust camera structure
- require semicolon after DISPLAY_ASSERT_EXCEPTION macro
- adjust s_game_cluster_bit_vectors structure and code that uses it
- use MASK macro in c_flags_no_init::valid
- adjust names in object_marker
- adjust call types for file reference functions so we can use INVOKE

Berthalamew committed:
remove redundant check in network_adapter_addresses_populate

Berthalamew committed:
rewrite screenshot_render function

- make sure the screenshot render target is only created if we're taking a screenshot, rather than every frame
- add screenshot_cubemap command

Kantanomo commented on xlive/Blam/Engine/cache/cache_files.cpp:
```diff
@@ -217,7 +217,10 @@ bool __cdecl scenario_tags_load(const char* scenario_path)
 
-	if(!cache_header_verify(cache_header) || strlen(cache_header->version_string) > 32)
+	if(!cache_header_verify(cache_header)
+		// This condition will never be true, included by bungie for some reason
```
This is essentially a sanity check if, for example if a cache header seems correct but there is data that exists where a null terminator should be the strlen will be invalid.. also could maybe just be a part of cache_header_verify.

Berthalamew replied:
readded

Berthalamew committed:
fix incorrect bitvector macro usage in object_reconnect_to_map

- changes BIT_VECTOR_SIZE_IN_BYTES instead of BIT_VECTOR_SIZE_IN_LONGS
- remove unused string in Config
- fix logic for find_first_sound_event and get_sound_reference_index

Berthalamew committed:
re-add length check

- use csstrnlen rather than strlen
- rename scenario_tags_load to scenario_tags_load_internal

Berthalamew committed:
fix crash in minidump code when trying to access uninitialized globals

- game_globals and game_options were accessed without checking if they were null first
- make private functions static
- adjust formatting
